calico_containers/tests/st/utils.py

```diff
@@ -17,14 +17,22 @@ def cleanup_inside(name):
     """
     Clean the inside of a container by deleting the containers and images within it.
     """
-    docker("exec", "-t", name, "bash", "-c",
-           "docker rm -f $(docker ps -qa) ; docker rmi $(docker images -qa)",
-           _ok_code=[0,
-                     1,  # Caused by 'docker: "rm" requires a minimum of 1 argument.' et al.
-                     127,  # Caused by '"docker": no command found'
-                     255,  # Caused by '"bash": executable file not found in $PATH'
-                    ]
-          )
+    try:
+        inner_containers = docker("exec", "-t", name, "bash", "-c", "docker ps -qa").stdout.split()
+        for container in inner_containers:
+            docker("exec", "-t", name, "bash", "-c", "docker rm -f %s" % container)
+
+        inner_images = docker("exec", "-t", name, "bash", "-c", "docker images -q").stdout.split()
+        for image in inner_images:
+            docker("exec", "-t", name, "bash", "-c", "docker rmi %s" % image)
+    except ErrorReturnCode_127:
```

Comment:
exception handling needs to be inside the for loops or you'll skip the rest of the cleanup if one rm fails.